evm/src/fixed_recursive_verifier.rs

```diff
@@ -1130,6 +1180,14 @@ where
                 nonzero_pis.insert(block_hashes_current_start + i, cur_targets[i]);
             }
 
+            // Initialize the checkpoint block number.
+            // Subtraction would result in an invalid proof for genesis, but we shouldn't try proving this block anyway.
```

Comment:
I'm not sure I understand this comment: if the genesis block is on the rhs, then the parent block has block number -1, but that also means that the lhs is a dummy block proof, right?

Reply:
Genesis being on the rhs means we're trying to prove it, but we can't.